### PR TITLE
[runtime] Cross32 changes.

### PR DESCRIPTION
**Pull request by vargaz**
Part of #9621

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/sgen/sgen-conf.h [hunk 1]
```diff
@@ -22,7 +22,14 @@ typedef guint32 mword;
 typedef guint64 mword;
 #endif
 
-typedef mword SgenDescriptor;
+#if TARGET_SIZEOF_VOID_P == 4
```
Comment (jaykrell) [inline mono/sgen/sgen-conf.h]: I would ask for a typedef somewhere of `target_gsize`, however that won't work with the single configure/compile/link goal.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
It’d be nice to say something like this:
```
case "$TARGET" in
*32* | ARM | X86 | POWERPC | MIPS | SPARC | WASM)
	target_sizeof_void_p=4
	;;
*64* | S390X)
	target_sizeof_void_p=8
	;;
*)
	AC_MSG_ERROR([unknown target])
	;;
esac
```

I think  it might work today even. We don't necessarily have a target compiler, just a host one I guess, so  things like MIPS n32 would have to be explicit on configure  command line.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build Linux ARMv5

**Comment by vargaz** [comment]
@monojenkins build Linux ARMv7 Interpreter

**Comment by vargaz** [comment]
Can somebody approve ?

**Pull request merged**